builtin/providers/aws/resource_aws_kinesis_analytics.go

```diff
@@ -0,0 +1,541 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kinesisanalytics"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func shim(*schema.ResourceData, interface{}) error {
+	return errors.New("func UPDATE Not implemented in resource_aws_kinesis_analytics.go")
+}
+
+func resourceAwsKinesisAnalytics() *schema.Resource {
+	return &schema.Resource{
+
+		Create: resourceAwsKinesisAnalyticsCreate,
+		Read:   resourceAwsKinesisAnalyticsRead,
+		Update: shim,
+		Delete: resourceAwskinesisAnalyticsDelete,
+
+		/*
+			todo:
+			some can trigger an Update instead of ForceNew
+			be sure to add docs about the json record type making some optional fields into required
+		*/
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"application_description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"application_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"inputs": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": { // NamePrefix
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"arn": { // KinesisStreamsInput.ResourceARN
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"role_arn": { // KinesisStreamsInput.RoleARN
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"record_format_type": { //RecordFormat.RecordFormatType ( JSON || CSV)
```

Comment:
If there are 2 allowed values, then we should validate them - look at validation.StringInSlice func